src/OrchardCore.Modules/Orchard.Liquid/Filters/TimeZoneFilter.cs

```diff
@@ -59,7 +59,7 @@ public TimeZoneFilter(ISiteService siteService, IClock clock)
 
             _localTimeZone = _localTimeZone ?? TimeZoneInfo.FindSystemTimeZoneById((await _siteService.GetSiteSettingsAsync()).TimeZone);
 
-            var local = TimeZoneInfo.ConvertTime(value, _localTimeZone);
+            var local = TimeZoneInfo.ConvertTime(value, _localTimeZone).LocalDateTime;
```

Comment:
This doesn't look correct. `LocalDateTime` will always be dependent of the current thread culture which we should never have to deal with